src/ManifoldsBase.jl

```diff
@@ -413,8 +413,15 @@ is `true`, the function either returns `true` or throws an error. By default the
 calls [`check_point(M, p; kwargs...)`](@ref) and checks whether the returned value
 is `nothing` or an error.
 """
-function is_point(M::AbstractManifold, p, throw_error = false; kwargs...)
-    mpe = check_point(M, p; kwargs...)
+function is_point(
+    M::AbstractManifold,
+    p,
+    throw_error = false;
+    atol::Real = 0,
+    rtol = atol > 0 ? 0 : sqrt(eps(number_eltype(p))),
```

Comment:
I'm not sure if that's the right default. What about something like
```julia
rtol = atol > 0 ? zero(number_eltype(p)) : default_rtol(M, p)
```
where `default_rtol(M, p)` would be by default more on the order of `sqrt(manifold_dimension(M))*eps(number_eltype(p))`? Or `prod(representation_size(M))` instead of `manifold_dimension(M)` if `representation_size(M)` is not nothing?

Reply:
I also was not sure, sure we can scale by the sqrt of the manifold dimension, on the other hand its the relative tolerance, so I thought the left hand side (being a quotient anyways) would cancel the manifold dimension out.

Reply:
My mental image for the proposed choice is that when points are represented using N numbers, then an epsilon change in every single one of them should, by default, also give a "close point". When each "coordinate" differs by eps(), then the Euclidean distance is sqrt(N)*eps().

Reply:
Hm, i am unsere here,m for me a relative error would mean that the sqrt(n) is “factored out” before that check anyways, but I will try to find out

Reply:
We could adapt the `rtoldefault` that Julia itself uses as wel? For me the norm implementation indicates that the Julia default does not do a sort(N) in the rtol: https://github.com/JuliaLang/julia/blob/7ecbfa9c926bdf380f824766b9d165ec9b133eaa/base/floatfuncs.jl#L313-L318

Reply:
Hm, you may be right but `eps(number_eltype(p))` seems too small and `sqrt(eps(number_eltype(p)))` seems too large. Very likely there is no reasonable manifold-independent default value.

Reply:
Remember that it is relative, so the manifold dimension (like vector dimensions in the classical case) should be taken care of.

Reply:
So do we keep the current state? Then I would just document it and this PR should be fine.

Reply:
I have one other minor point about type stability but leaving sqrt(eps) here seems to be the best option.

Reply:
I've just made a comment below about it :slightly_smiling_face: .